plugins/modules/ndo_port_channel_interface.py

```diff
@@ -0,0 +1,351 @@
+        type: dict
+        suboptions:
+            name:
+                description:
+                - The name of the Port Channel Interface Setting Policy.
+                type: str
+            template:
+                description:
+                - The name of the template in which is referred the Port Channel Interface Policy Group.
+                type: str
+        aliases: [ policy, interface_policy ]
+    interface_descriptions:
+        description:
+        - The list of descriptions for each interface.
+        - The interface must exist when a description is provided.
+        type: list
+        elements: dict
+        suboptions:
+            interface_id:
+                description:
+                - The interface ID.
+                type: str
+            description:
+                description:
+                - The description of the interface.
+                type: str
+    state:
+        description:
+        - Use C(absent) for removing.
+        - Use C(query) for listing an object or multiple objects.
+        - Use C(present) for creating or updating.
+        type: str
+        choices: [ absent, query, present ]
+        default: query
+extends_documentation_fragment: cisco.mso.modules
+"""
+
+EXAMPLES = r"""
+- name: Create a new Port Channel Interface
+  cisco.mso.ndo_port_channel_interface:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_fabric_resource_template
+    description: My Ansible Port Channel
+    port_channel_interface: ansible_port_channel_interface
+    node: 101
+    interfaces:
+      - 1/1
+      - 1/10-11
+    interface_policy_group:
+      name: ansible_policy_group
+      template: ansible_fabric_policy_template
+    interface_descriptions:
+      - interface_id: 1/1
+        description: My first Ansible Interface
+      - interface_id: 1/10
+        description: My second Ansible Interface
+      - interface_id: 1/11
+        description: My third Ansible Interface
+    state: present
+
+- name: Query a Port Channel Interface with template_name
+  cisco.mso.ndo_port_channel_interface:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_fabric_resource_template
+    port_channel_interface: ansible_port_channel_interface
+    state: query
+  register: query_one
+
+- name: Query all Port Channel Interfaces in the template
+  cisco.mso.ndo_port_channel_interface:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_fabric_resource_template
+    state: query
+  register: query_all
+
+- name: Delete a Port Channel Interface
+  cisco.mso.ndo_port_channel_interface:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_fabric_resource_template
+    port_channel_interface: ansible_port_channel_interface
+    state: absent
+"""
```

Comment:
Add examples for query with UUID and delete with UUID

Reply:
will add them